altair/utils/data.py

```diff
@@ -350,14 +345,27 @@ def _compute_data_hash(data_str: str) -> str:
     return hashlib.sha256(data_str.encode()).hexdigest()[:32]
 
 
+def _from_geo_interface(data: SupportsGeoInterface | Any) -> dict[str, Any]:
+    """
+    Santize a ``__geo_interface__`` w/ pre-santize step for ``pandas`` if needed.
+
+    Notes
+    -----
+    Split out to resolve typing issues related to:
+    - Intersection types
+    - ``typing.TypeGuard``
+    - ``pd.DataFrame.__getattr__``
+    """
+    if _is_pandas_dataframe(data):
+        data = sanitize_pandas_dataframe(data)
+    return sanitize_geo_interface(data.__geo_interface__)
```

Comment:
@MarcoGorelli I've somewhat bailed on the typing for this. 
IMO there's too many interacting things that it all just becomes noise.

Hopefully with this split into a new function, no one needs to worry about that again 😅

Reply:
nice one, thanks @dangotbanned !